compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaArrayTypeSymbol.java

```diff
@@ -49,6 +49,9 @@ public TypeSymbol memberTypeDescriptor() {
 
     @Override
     public String signature() {
+        if (memberTypeDescriptor().typeKind() == TypeDescKind.UNION) {
```

Comment:
Shall we introduce a `ParanthesizedTypeDescriptor`, so we don't need to special case the union here?

Reply:
As discussed, this could also appear without parenthesis as well. hence we need this.
```ballerina
type Value string|int|float;
Value[]|error myValue = [];
```